interface/main/calendar/modules/PostCalendar/pnuserapi.php

```diff
@@ -881,6 +883,12 @@ function &postcalendar_userapi_pcQueryEvents($args)
     "(a.pc_endDate = '0000-00-00' AND a.pc_eventDate >= '" . pnVarPrepForStore($start) . "' AND " .
     "a.pc_eventDate <= '" . pnVarPrepForStore($end) . "')) ";
 
+    // Custom filtering
+    $calFilterEvent = new CalendarFilterEvent();
+    $calFilterEvent = $GLOBALS["kernel"]->getEventDispatcher()->dispatch(CalendarFilterEvent::EVENT_HANDLE, $calFilterEvent, 10);
```

Comment:
What happens if there is more than one event listener on these dispatch calls?

Reply:
@bradymiller Like Ken said down below if there are more than one event listener that is listening for this specific event the event object (CalendarFilterEvent) will get sent to each listener and can be modified or returned by the subsequent listener.  Each listener when they listen to the event can set a priority to move themselves up or down the chain of calls. For example you have 4 listeners that are added in the following order:

Listener A (no priority set, defaults to 10)
Listener B (no priority set, defaults to 10)
Listener C (priority set to 0)
Listener D (Priority set to 5)

Then when this event called CalendarFilterEvent.A the following listener sequence will occur:
Listener C (with priority 0) called with CalendarFilterEvent.A.  It modifies the event and returns it
Listener D (with priority 5) called with CalendarFilterEvent.A.  It does nothing with the event and returns CalendarFilterEvent.B.
Listener A (with priority 10) called with CalendarFilterEvent.A. It throws away the event and returns a new event CalendarFilterEvent.B.  
Listener B (with priority 10) called with CalendarFilterEvent.B.  It modifies the event and returns it.

The value of $calFilterEvent is now CalendarFilterEvent.B.

Reply:
thanks for the explanation @adunsulag . very helpful!